src/All2allModelRegion.cpp

```diff
@@ -102,40 +110,52 @@ namespace geopm
+                    ModelRegion::loop_enter(i);
 
-                double timeout = 0.0;
-                struct geopm_time_s start = {{0,0}};
-                struct geopm_time_s curr = {{0,0}};
-                int loop_done = 0;
-                if (!m_rank) {
-                    (void)geopm_time(&start);
-                }
-                while (!loop_done) {
-                    int err = MPI_Alltoall(m_send_buffer, m_num_send, MPI_CHAR, m_recv_buffer,
-                                           m_num_send, MPI_CHAR, MPI_COMM_WORLD);
-                    if (err) {
-                        throw Exception("MPI_Alltoall()", err, __FILE__, __LINE__);
-                    }
+                    double timeout = 0.0;
+                    struct geopm_time_s start = {{0,0}};
+                    struct geopm_time_s curr = {{0,0}};
+                    int loop_done = 0;
                     if (!m_rank) {
-                        (void)geopm_time(&curr);
-                        timeout = geopm_time_diff(&start, &curr);
-                        if (timeout > (m_big_o / m_num_progress_updates)) {
-                            loop_done = 1;
-                        }
+                        (void)geopm_time(&start);
                     }
-                    err = MPI_Bcast((void*)&loop_done, 1, MPI_INT, 0, MPI_COMM_WORLD);
-                    if (err) {
-                        throw Exception("MPI_Bcast()", err, __FILE__, __LINE__);
+                    while (!loop_done) {
+                        int err = 0;
+                        err = MPI_Alltoall(m_send_buffer, m_num_send, MPI_CHAR, m_recv_buffer,
+                                           m_num_send, MPI_CHAR, MPI_COMM_WORLD);
+                        if (err) {
+                            throw Exception("MPI_Alltoall()", err, __FILE__, __LINE__);
+                        }
+                        if (!m_rank) {
+                            (void)geopm_time(&curr);
+                            timeout = geopm_time_diff(&start, &curr);
+                            if (timeout > (m_big_o / m_num_progress_updates)) {
+                                loop_done = 1;
+                            }
+                        }
+                        err = MPI_Bcast((void*)&loop_done, 1, MPI_INT, 0, MPI_COMM_WORLD);
+                        if (err) {
+                            throw Exception("MPI_Bcast()", err, __FILE__, __LINE__);
+                        }
                     }
-                }
 
-                ModelRegion::loop_exit();
+                    ModelRegion::loop_exit();
+                }
+                ModelRegion::region_exit();
+            }
+            else {
+                ModelRegion::region_enter();
+                for (uint64_t i = 0; i < m_num_progress_updates; ++i) {
+                    ModelRegion::loop_enter(i);
+                    std::this_thread::sleep_for(std::chrono::microseconds(100));
```

Comment:
geopmbench.1.rst should probably be updated to say that all2all becomes sleep when MPI is disabled.

We should document all the changed behaviors for the other regions as applicable.